File: GradeTrackerAPI/Entities/EducationTypeGoal.cs
namespace GradeTrackerAPI.Entities
{
    public class EducationTypeGoal : BaseEntity
    {
        [Column(TypeName = "decimal(3,2)")]
        public double? AverageDesiredMark { get; set; }
        [ForeignKey(nameof(EducationTypeId))]
        public int? EducationTypeId { get; set; }
        public virtual EducationType? EducationType { get; set; }
        [ForeignKey(nameof(UserId))]
        public int? UserId { get; set; }
        public virtual User? User { get; set; }
    }
}

File: GradeTracker.Client/src/app/_core/models/educationTypeGoal.ts
import { User } from "./user"
import { EducationType } from "./educationType"

export class EducationTypeGoal {
    id?: number;
    averageDesiredMark?: number;
    userId? : number;
    user?: User;
    educationTypeId?: number;
    educationType?: EducationType;
}